yaptide/celery/tasks.py

```diff
@@ -134,29 +136,32 @@ def read_file(stats: SimulationStats, filepath: Path, task_id: int):
 
 
 @celery_app.task(bind=True)
-def run_simulation(self, json_data: dict):
+def run_simulation(self, payload_dict: dict):
     """Simulation runner"""
     # create temporary directory
     with tempfile.TemporaryDirectory() as tmp_dir_path:
+        runner_obj = SHRunner(jobs=payload_dict.get("ntasks"),
+                              keep_workspace_after_run=True,
+                              output_directory=tmp_dir_path)
+
+        # since it is not obligatory for UI to provide ntasks parameters it is set here for sure
+        # if it is provided it will be overwritten by itself so nothing will change
+        payload_dict["ntasks"] = runner_obj.jobs
```

Comment:
Its not a good practice that `run_simulation` method modifies its argument (`payload_dict`)